src/pages/Pools.tsx

```diff
@@ -283,58 +100,73 @@ function Pools(): ReactElement | null {
+                  basicPool.isPaused)),
           )
-          .sort(([a, aIsOutdated], [b, bIsOutdated]) => {
+          .sort((a, b) => {
             // 1. user pools
             // 2. active pools
             // 3. higher TVL pools
-            if (
-              (a.userShareData?.usdBalance || Zero).gt(Zero) ||
-              (b.userShareData?.usdBalance || Zero).gt(Zero)
-            ) {
-              return (a.userShareData?.usdBalance || Zero).gt(
-                b.userShareData?.usdBalance || Zero,
-              )
-                ? -1
-                : 1
+            const userLpTokenBalanceA =
+              userState?.tokenBalances?.[a.lpToken] || Zero
+            const userLpTokenBalanceB =
+              userState?.tokenBalances?.[b.lpToken] || Zero
+            const poolAssetA = parseUnits(
+              String(
+                tokenPricesUSD?.[getTokenSymbolForPoolType(a.typeOfAsset)] || 0,
+              ),
+              18,
+            )
+            const poolAssetB = parseUnits(
+              String(
+                tokenPricesUSD?.[getTokenSymbolForPoolType(b.typeOfAsset)] || 0,
+              ),
+              18,
+            )
+            const userBalanceUSDA = userLpTokenBalanceA
+              .mul(poolAssetA)
+              .div(BigNumber.from(BigInt(1e18)))
+            const userBalanceUSDB = userLpTokenBalanceB
+              .mul(poolAssetB)
+              .div(BigNumber.from(BigInt(1e18)))
+            const poolTVLUSDA = a.lpTokenSupply
+              .mul(poolAssetA)
+              .div(BigNumber.from(BigInt(1e18)))
+            const poolTVLUSDB = b.lpTokenSupply
+              .mul(poolAssetB)
+              .div(BigNumber.from(BigInt(1e18)))
+            const isOutdatedA = a.isMigrated || a.isGuarded || a.isPaused
+            const isOutdatedB = b.isMigrated || b.isGuarded || b.isPaused
+            if (userBalanceUSDA.gt(Zero) || userBalanceUSDB.gt(Zero)) {
+              return userBalanceUSDA.gt(userBalanceUSDB) ? -1 : 1
             } else if (
-              a.poolData.isMigrated ||
-              b.poolData.isMigrated ||
-              aIsOutdated ||
-              bIsOutdated
+              !(isOutdatedA && isOutdatedB) &&
+              (isOutdatedA || isOutdatedB)
             ) {
-              return a.poolData.isMigrated || aIsOutdated ? 1 : -1
+              return isOutdatedA ? 1 : -1
             } else {
-              return (a.poolData?.reserve || Zero).gt(
-                b.poolData?.reserve || Zero,
-              )
-                ? -1
-                : 1
+              return poolTVLUSDA.gt(poolTVLUSDB) ? -1 : 1
             }
           })
-          .map(([poolProps]) => (
+          .map((basicPool) => (
             <PoolOverview
-              key={poolProps.name}
-              {...poolProps}
+              key={basicPool.poolName}
+              poolName={basicPool.poolName}
+              poolRoute={`/pools/${basicPool.poolName}`} // TODO
```

Comment:
should be fine, decode later if param value needs to be used somewhere.

Reply:
I think the risk of using raw pool names (in urls, rendering to page) is that people are sneaky and will name their pools shit like `<script>alert("🖕🖕🖕")</script>`

Reply:
Curve seems to determine the route path off some number. I wonder if name will scale.

Reply:
yeah they use, I believe, the pool's index in the registry for factory pools. Seems prudent. I think address or id are safe, name is trickier.

Reply:
one other thing to mention - we'll need to keep some static routes to maintain backwards compatibility in case partners are linking to us

Reply:
We could use a URL cleansing library or something to prevent this.

Reply:
A dom sanitizer yeah. Would be cool to have 1:1 relationship with poolname on FE and on contract.